edge-modules/ModuleRestarter/src/Program.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace ModuleRestarter
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class Program
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("ModuleRestarter");
+
+        public static int Main() => MainAsync().Result;
+
+        static async Task<int> MainAsync()
+        {
+            Logger.LogInformation($"Starting module restarter with the following settings:\r\n{Settings.Current}");
+
+            if (Settings.Current.DesiredModulesToRestart.Count == 0)
+            {
+                Logger.LogInformation("No modules names found in input. Stopping.");
+                return 0;
+            }
+
+            (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
+
+            await RestartModules(cts);
+
+            completed.Set();
+            handler.ForEach(h => GC.KeepAlive(h));
+            Logger.LogInformation("ModuleRestarter Main() finished.");
+            return 0;
+        }
+
+        /// <summary>
+        /// Restarts random modules periodically (with default restart occurrence once every 10 minutes)
+        /// </summary>
+        static async Task RestartModules(CancellationTokenSource cts)
+        {
+            ServiceClient iotHubServiceClient = ServiceClient.CreateFromConnectionString(Settings.Current.ServiceClientConnectionString);

```

Comment:
this line may throw exception; please handle it.

Reply:
I looked at the implementation within DirectMethodCloudSender. Two things I would like to confirm:
- Since this is a cloud operation, we need to retry in case of transient failure. DMCS exits the application, however their restartPolicy is set to 'always' which ensures a retry. We are going to have to manually retry this.
- DMCS nests try catches which I feel like we should not do. I am going to try catch this separately.